src/VideoDepacketizer.c

```diff
@@ -230,7 +230,8 @@ static bool isIdrFrameStart(PBUFFER_DESC buffer) {
     return getSpecialSeq(buffer, &specialSeq) &&
         isSeqFrameStart(&specialSeq) &&
         (specialSeq.data[specialSeq.offset + specialSeq.length] == 0x67 || // H264 SPS
-         specialSeq.data[specialSeq.offset + specialSeq.length] == 0x40); // H265 VPS
+         specialSeq.data[specialSeq.offset + specialSeq.length] == 0x40 || // H265 VPS
+         specialSeq.data[specialSeq.offset + specialSeq.length + 7] == 0x40); // HEVC encoded by AMF
```

Comment:
I don't think it's safe to do this `+ 7` here. For regular P-frame NALs, this will be accessing data that is undefined (video content) that could incorrectly cause `isIdrFrameStart()` to return true.

There is also other code here which may make incorrect assumptions when it receives a AUD NAL like `getBufferFlags()`.

I think it would probably be better to strip the AUD NAL out as early as possible, like right after this point: https://github.com/moonlight-stream/moonlight-common-c/blob/a64af8d8cf1e760759fba6026b10c6d4a5f2f3d7/src/VideoDepacketizer.c#L602-L606

Reply:
I'll think about it a little